src/prpy/planning/base.py

```diff
@@ -52,13 +52,27 @@ def __init__(self, func):
 
     def __call__(self, instance, robot, *args, **kw_args):
         env = robot.GetEnv()
-
-        with Clone(env, clone_env=instance.env):
-            planning_traj = self.func(instance, Cloned(robot), *args, **kw_args)
-            traj = openravepy.RaveCreateTrajectory(env, planning_traj.GetXMLId())
-            traj.Clone(planning_traj, 0)
-
-        return traj
+        defer = kw_args.get('defer')
+
+        with Clone(env, clone_env=instance.env, lock=True, unlock=False) \
+                as cloned_env:
+
+            def call_planner():
+                planning_traj = self.func(instance, Cloned(robot),
```

Comment:
This looks like a potential race condition. `Cloned()` is only valid while in the `with` block. Since `call_planner` is called by the executor, it may not be executed until after we `__exit__` the `with`-block. To be safe, you should resolve the `Cloned()` call outside the `call_planner()` function, but inside the `with`-block.